spec/controllers/stories_controller_spec.rb

```diff
@@ -71,52 +57,84 @@
+      end
+
+      it "redirects to the edit page" do
+        expect(response).to render_template :edit
+      end
+
+      it "shows the fields for the story" do
+        expect(assigns(:story)).to eq story
+      end
     end
-  end
 
-  describe "#update" do
-    it "updates the story" do
-      put :update, params: {id: story.id,
-                            project_id: project.id,
-                            story: {title: "New Story"}}
+    describe "#update" do
+      it "updates the story" do
+        put :update, params: {id: story.id,
+                              project_id: project.id,
+                              story: {title: "New Story"}}
 
-      expect(story.reload.title).to eq "New Story"
+        expect(story.reload.title).to eq "New Story"
+      end
     end
-  end
 
-  describe "#bulk_destroy" do
-    it "deletes multiple stories" do
-      ids = [99, 100]
-      ids.each do |id|
-        FactoryBot.create(:story, id: id, project: project)
+    describe "#bulk_destroy" do
+      it "deletes multiple stories" do
+        ids = [99, 100]
+        ids.each do |id|
+          FactoryBot.create(:story, id: id, project: project)
+        end
+
+        expect {
+          post :bulk_destroy, params: {ids: ids}, format: :json
+        }.to change(Story, :count).by(-2)
       end
+    end
 
-      expect {
-        post :bulk_destroy, params: {ids: ids}, format: :json
-      }.to change(Story, :count).by(-2)
+    describe "#move" do
+      it "does not allow moving stories to non-sibling projects" do
+        project2 = FactoryBot.create(:project, parent: project)
+        project3 = FactoryBot.create(:project)
+        story = FactoryBot.create(:story, project: project2)
+
+        put :move, params: {project_id: project2.id, story_id: story.id, to_project: project3.id}
+        expect(flash[:error]).to eq "Selected project does not exist or is not a sibling."
+        expect(response).to redirect_to project2
+      end
     end
   end
 
-  describe "#move" do
-    it "does not allow moving stories to non-sibling projects" do
-      project2 = FactoryBot.create(:project, parent: project)
-      project3 = FactoryBot.create(:project)
-      story = FactoryBot.create(:story, project: project2)
+  context "when the story project is archived" do
+    before do
+      story.project.toggle_archived!
+    end
+
+    it "doesn't allow me to edit the story" do
+      get :edit, params: {id: story.id, project_id: project.id}
+      expect(response).to redirect_to project_path(project)
+    end
 
-      put :move, params: {project_id: project2.id, story_id: story.id, to_project: project3.id}
-      expect(flash[:error]).to eq "Selected project does not exist or is not a sibling."
-      expect(response).to redirect_to project2
+    it "doesn't allow me to update the story" do
+      put :update, params: {id: story.id, project_id: project.id, story: {title: "Changed Title"}}
+      expect(response).to redirect_to project_path(project)
```

Comment:
same here, a successful update on a story also redirects to project_path(project), we can check that title didn't change to verify the update was prevented